test/EIP20CoGateway/_redeemRawTx.js

```diff
@@ -0,0 +1,212 @@
+        redeemParams.hashLock,
+      ),
+      expectedErrorMessage,
+    );
+  });
+
+  it('should throw error when beneficiary address is invalid', async () => {
+    const expectedErrorMessage = `Invalid beneficiary address: 0x123.`;
+
+    await AssertAsync.reject(
+      coGateway._redeemRawTx(
+        redeemParams.amount,
+        `0x123`,
+        redeemParams.gasPrice,
+        redeemParams.gasLimit,
+        redeemParams.nonce,
+        redeemParams.hashLock,
+      ),
+      expectedErrorMessage,
+    );
+  });
+
+  it('should throw error when gas price is undefined', async () => {
+    const expectedErrorMessage = `Invalid gas price: ${undefined}.`;
+
+    await AssertAsync.reject(
+      coGateway._redeemRawTx(
+        redeemParams.amount,
+        redeemParams.beneficiary,
+        undefined,
+        redeemParams.gasLimit,
+        redeemParams.nonce,
+        redeemParams.hashLock,
+      ),
+      expectedErrorMessage,
+    );
+  });
+
+  it('should throw error when gas limit is undefined', async () => {
+    const expectedErrorMessage = `Invalid gas limit: ${undefined}.`;
+
+    await AssertAsync.reject(
+      coGateway._redeemRawTx(
+        redeemParams.amount,
+        redeemParams.beneficiary,
+        redeemParams.gasPrice,
+        undefined,
+        redeemParams.nonce,
+        redeemParams.hashLock,
+      ),
+      expectedErrorMessage,
+    );
+  });
+
+  it('should throw error when nonce is undefined', async () => {
+    const expectedErrorMessage = `Invalid nonce: ${undefined}.`;
+
+    await AssertAsync.reject(
+      coGateway._redeemRawTx(
+        redeemParams.amount,
+        redeemParams.beneficiary,
+        redeemParams.gasPrice,
+        redeemParams.gasLimit,
+        undefined,
+        redeemParams.hashLock,
+      ),
+      expectedErrorMessage,
+    );
+  });
+
+  it('should throw error when nonce is undefined', async () => {
+    const expectedErrorMessage = `Invalid hash lock: ${undefined}.`;
+
+    await AssertAsync.reject(
+      coGateway._redeemRawTx(
+        redeemParams.amount,
+        redeemParams.beneficiary,
+        redeemParams.gasPrice,
+        redeemParams.gasLimit,
+        redeemParams.nonce,
+        undefined,
+      ),
+      expectedErrorMessage,
+    );
+  });
+
+  it('should return correct mocked transaction object', async () => {
+    setup();
+    const result = await coGateway._redeemRawTx(
+      redeemParams.amount,
+      redeemParams.beneficiary,
+      redeemParams.gasPrice,
+      redeemParams.gasLimit,
+      redeemParams.nonce,
+      redeemParams.hashLock,
+    );
+    assert.strictEqual(
+      result,
+      mockedTx,
+      'Function should return mocked transaction object.',
+    );
+
+    SpyAssert.assert(spyMethod, 1, [
+      [
+        redeemParams.amount,
+        redeemParams.beneficiary,
+        redeemParams.gasPrice,
+        redeemParams.gasLimit,
+        redeemParams.nonce,
+        redeemParams.hashLock,
+      ],
+    ]);
```

Comment:
Can there be an error that gets printed when the assert fails?

Reply:
There is already an error for param count mismatch and param value mismatch in `SpyAssert.assert`

Reply:
Created #90

Reply:
👍